### PR TITLE
[CoW Subsidy] Base PR

### PR DESCRIPTION
**Pull request by W3stside**
# Summary

Part of <TBD> 

CoW Subsidy modal!

image

TODO (upcoming PRs)

- [x] re-enable account prerequisite (currently inverted for testing) - incoming #2551 
- [ ] use real balance (currently mocked) - dependent on #2552 
- [x] add CTA (currently always showing) - incoming #2551 
- [x] add fee subsidy comp in swap mod - incoming #2551 
- [x] Add current TIER highlighting component - incoming #2558 
- [x] Add function/logic for getting current tier from user's balance - incoming #2557 
- [ ] @fairlighteth styling
- [ ] switch columns: DISCOUNT - THRESHOLD

### CODE REVIEW

**Comment by elena-zh** [comment]
Seems that the build has failed.

File: src/custom/components/CowBalance/index.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    <Wrapper>
+      <CowProtocolLogo size={100} />
+
+      {title && (
+        <ClaimSummaryTitle>
+          <Trans>{title}</Trans>
+        </ClaimSummaryTitle>
+      )}
+
+      {balance && !account && (
```
Comment (alfetopito) [inline src/custom/components/CowBalance/index.tsx:31]: Why `!account`?

Ahh ok, there's a comment in the PR body.

Can you add a TODO or comment to be explicit?

File: src/custom/components/CowSubsidyModal/index.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+)
+
+export default function CowSubsidyModal({
+  isOpen,
+  onDismiss,
+  ...restProps
+}: Pick<ConfirmationModalProps, 'isOpen'> & Omit<ConfirmationModalContentProps, 'title' | 'topContent'>) {
+  const { chainId } = useActiveWeb3React()
+
+  const { activeClaimAccount } = useClaimState()
```
[the remark this answers is not in the document]
Comment (alfetopito) [reply src/custom/components/CowSubsidyModal/index.tsx:48]: Got it, updated in the next PR, disregard

**Review by alfetopito — APPROVED** [review]

File: src/custom/constants/subsidy.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+export const COW_SUBSIDY = [
+  ['(v)COW balance', 'Fee discount'],
```
Comment (anxolin) [inline src/custom/constants/subsidy.ts:2]: I find it strange to add the headers here, why is this?  Can this be part of the content together with the other content instead fo here mixed with the data?

File: src/custom/constants/subsidy.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+  ['0', 0],
+  ['100', 10],
+  ['1000', 15],
+  ['10000', 25],
+  ['100000', 35],
+]
+
+export const SUBSIDY_INFO_MESSAGE = 'As a (v)COW token holder you will be eligbible for a fee discount.'
```
Comment (anxolin) [inline src/custom/constants/subsidy.ts:10]: Same as this, this is content. Shouldn't be in a constant imo. I know, u want to refactor some repetition. I don't think a bit of repetition in content is that bad, and if its in a constant should be in a constant in the component, not a global one

Comment (anxolin) [inline src/custom/constants/subsidy.ts:10]: the message has a typo 📕 "eligBible'

File: src/custom/components/CowSubsidyModal/SubsidyTable.tsx [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+      <thead>
+        <SubsidyTr>
+          {headers.map((header, i) => (
+            <th key={header + '_' + i}>{header}</th>
+          ))}
+        </SubsidyTr>
+      </thead>
+      <tbody>
+        {body.map(([threshold, discount], i) => (
+          <SubsidyTr key={discount + '_' + i}>
```
Comment (anxolin) [inline src/custom/components/CowSubsidyModal/SubsidyTable.tsx:36]: why key is not just i, why concatenating?

Comment (anxolin) [reply src/custom/components/CowSubsidyModal/SubsidyTable.tsx:36]: same as above

File: src/custom/components/CowSubsidyModal/SubsidyTable.tsx [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    font-weight: 300;
+  }
+  > td,
+  th {
+    padding: 10px;
+    text-align: center;
+  }
+`
+
+function SubsidyTable({ data }: { data: (string | number)[][] }) {
```
Comment (anxolin) [inline src/custom/components/CowSubsidyModal/SubsidyTable.tsx:21]: if you define the data as just data and not include the headers, is better model and types are also simpler

File: src/custom/components/CowSubsidyModal/index.tsx [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+export default function CowSubsidyModal({
+  isOpen,
+  onDismiss,
+  ...restProps
+}: Pick<ConfirmationModalProps, 'isOpen'> & Omit<ConfirmationModalContentProps, 'title' | 'topContent'>) {
+  const { account, chainId } = useActiveWeb3React()
+
+  const vCow = chainId ? V_COW[chainId] : undefined
+  // TODO: vcow and cow balance from @nenadV91
```
Comment (anxolin) [inline src/custom/components/CowSubsidyModal/index.tsx:41]: 💪

File: src/custom/components/CowSubsidyModal/index.tsx [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+import SubsidyTable from './SubsidyTable'
+
+const CowSubsidyInfo = ({ account, balance }: CowBalanceProps) => (
+  <AutoColumn style={{ marginTop: 20, padding: '2rem 0' }} gap="24px" justify="center">
+    <Text fontWeight={500} fontSize={16} style={{ textAlign: 'center', width: '85%', wordBreak: 'break-word' }}>
+      {SUBSIDY_INFO_MESSAGE}
+    </Text>
+    {/* VCOW LOGO */}
+    <CowBalance account={account} balance={balance} />
+    <SubsidyTable data={COW_SUBSIDY} />
```
Comment (anxolin) [inline src/custom/components/CowSubsidyModal/index.tsx:29]: you should pass the balance too, right? Just to highlight the row

File: src/custom/pages/Swap/SwapMod.tsx [hunk 9]
```diff
@@ -169,6 +171,9 @@ export default function Swap({
     [setTransactionConfirmationModalMsg, openTransactionConfirmationModalAux]
   )
 
+  // Cow subsidy modal
+  const openCowSubsidyModal = useOpenModal(ApplicationModal.COW_SUBSIDY)
+  const showCowSubsidyModal = useModalOpen(ApplicationModal.COW_SUBSIDY)
```
Comment (anxolin) [inline src/custom/pages/Swap/SwapMod.tsx:176]: 🥲 every time we add new stuff to `SwapMod` god kills a kitten

wen refactor :) 

obviously not this PR, only kidding ~kinda

**Review by anxolin — APPROVED** [review]
Nice progress! I left a bunch of comments, but is an approval. I think we should merge and keep iterating in new PRs and addressing comments

File: src/custom/pages/Swap/SwapMod.tsx (hunk 9 above)
Comment (alfetopito) [reply src/custom/pages/Swap/SwapMod.tsx:176]: 🥲

File: src/custom/components/CowSubsidyModal/index.tsx (hunk 8 above)
Comment (W3stside) [reply src/custom/components/CowSubsidyModal/index.tsx:29]: incoming in another pr these were split

File: src/custom/components/CowSubsidyModal/constants.ts [hunk 10]
```diff
@@ -0,0 +1,13 @@
+// used in modal and popover
+export const SUBSIDY_INFO_MESSAGE = 'As a (v)COW token holder you will be eligbible for a fee discount.'
```
Comment (W3stside) [inline src/custom/components/CowSubsidyModal/constants.ts:2]:
```suggestion
export const SUBSIDY_INFO_MESSAGE = 'As a (v)COW token holder you will be eligible for a fee discount.'
```

**Pull request merged**